src/snowflake/snowpark/modin/pandas/dataframe.py

```diff
@@ -1794,23 +1794,31 @@ def nsmallest(self, n, columns, keep="first"):  # noqa: PR01, RT01, D200
             )
         )
 
-    @dataframe_not_implemented()
-    def unstack(self, level=-1, fill_value=None):  # noqa: PR01, RT01, D200
+    def unstack(
+        self,
+        level: int | str | list = -1,
+        fill_value: int | str | dict = None,
+        sort: bool = True,
+    ):
         """
         Pivot a level of the (necessarily hierarchical) index labels.
         """
         # TODO: SNOW-1063346: Modin upgrade - modin.pandas.DataFrame functions
-        if not isinstance(self.index, pandas.MultiIndex) or (
-            isinstance(self.index, pandas.MultiIndex)
-            and is_list_like(level)
-            and len(level) == self.index.nlevels
+        # This ensures that non-pandas MultiIndex objects are caught.
+        is_multiindex = len(self.index.names) > 1
```

Comment:
This check in Modin results in a query being issued to pull the index -- I wonder if we deviate from Modin temporarily and push this down to the query compiler as we have to do another index check there (and we can reduce to 2 queries in total).

Reply:
There is a query compiler method supported in OSS Modin that is `self._query_compiler.has_multiindex` that you could use here. Example in Modin: 

https://github.com/modin-project/modin/blob/759d548814a6ac224e83e7531cf98e20b13d85cb/modin/pandas/dataframe.py#L3227